source/shared/cpp/ObjectModel/CollectionTypeElement.cpp

```diff
@@ -0,0 +1,114 @@
+#include "pch.h"
+#include "CollectionTypeElement.h"
+#include "Util.h"
+
+using namespace AdaptiveSharedNamespace;
+
+CollectionTypeElement::CollectionTypeElement(CardElementType type, ContainerStyle style, 
+    VerticalContentAlignment alignment) : BaseCardElement(type), m_style(style), 
+    m_verticalContentAlignment(alignment), m_hasPadding(false), m_hasBleed(false), m_parentalId()
+{
+}
+
+ContainerStyle CollectionTypeElement::GetStyle() const
+{
+    return m_style;
+}
+
+void CollectionTypeElement::SetStyle(const ContainerStyle value)
+{
+    m_style = value;
+}
+
+VerticalContentAlignment CollectionTypeElement::GetVerticalContentAlignment() const
+{
+    return m_verticalContentAlignment;
+}
+
+void CollectionTypeElement::SetVerticalContentAlignment(const VerticalContentAlignment value)
+{
+    m_verticalContentAlignment = value;
+}
+
+bool CollectionTypeElement::GetPadding() const
+{
+    return m_hasPadding;
+}
+
+void CollectionTypeElement::SetPadding(const bool value)
+{
+    m_hasPadding = value;
+}
+
+// Applies Padding Flag When appropriate
+void CollectionTypeElement::ConfigPadding(const ParseContext& context)
+{
+    // we set padding when parental style is different from child's, and its style should not be None
+    if(GetStyle() != ContainerStyle::None) 
+    {
+        if(context.GetParentalContainerStyle() != GetStyle())
```

Comment:
Maybe consolidate these:

    if(GetStyle() != ContainerStyle::None && 
       context.GetParentalContainerStyle() != GetStyle())

Reply:
the above won't cover case when parent style is None and the current's style is Default. In that case, the current doesn't get padding.

---
In reply to: [258670492](https://github.com/Microsoft/AdaptiveCards/pull/2406#discussion_r258670492) [](ancestors = 258670492)